docker-compose.yml

```diff
@@ -1,133 +1,159 @@
+    depends_on:
+      - db
+      - redis
+      - es
 
   streaming:
-    build: .
-    image: ghcr.io/mastodon/mastodon:v4.2.9
-    restart: always
+    # build:
+    #   context: ./streaming
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon-streaming:nightly
     env_file: .env.production
-    command: node ./streaming
+    environment:
+      - PORT=4000
+      - BIND=127.0.0.1
+    restart: always
+    command: node ./streaming/index.js
+    healthcheck:
+      test:
+        [
+          'CMD-SHELL',
+          "curl -s --noproxy localhost localhost:4000/api/v1/streaming/health | grep -q 'OK' || exit 1",
+        ]
     networks:
       - external_network
       - internal_network
-    healthcheck:
-      # prettier-ignore
-      test: ['CMD-SHELL', 'wget -q --spider --proxy=off localhost:4000/api/v1/streaming/health || exit 1']
     ports:
       - '127.0.0.1:4000:4000'
     depends_on:
       - db
       - redis
 
-  sidekiq:
-    build: .
-    image: ghcr.io/mastodon/mastodon:v4.2.9
+  db:
+    image: postgres:16-alpine
     restart: always
-    env_file: .env.production
-    command: bundle exec sidekiq
-    depends_on:
-      - db
-      - redis
+    healthcheck:
+      test: ['CMD', 'pg_isready', '-U', 'postgres']
+    shm_size: 256mb
     networks:
-      - external_network
       - internal_network
     volumes:
-      - ./public/system:/mastodon/public/system
+      - ./postgres16:/var/lib/postgresql/data
+    environment:
+      - 'POSTGRES_HOST_AUTH_METHOD=trust'
+
+  redis:
+    restart: always
+    image: redis:7-alpine
     healthcheck:
-      test: ['CMD-SHELL', "ps aux | grep '[s]idekiq\ 6' || false"]
+      test: ['CMD', 'redis-cli', 'ping']
+    networks:
+      - internal_network
+    volumes:
+      - ./redis:/data
+
+  es:
+    image: docker.elastic.co/elasticsearch/elasticsearch:7.17.21
+    environment:
+      - 'ES_JAVA_OPTS=-Xms512m -Xmx512m -Des.enforce.bootstrap.checks=true'
+      - 'xpack.license.self_generated.type=basic'
+      - 'xpack.security.enabled=false'
+      - 'xpack.watcher.enabled=false'
+      - 'xpack.graph.enabled=false'
+      - 'xpack.ml.enabled=false'
+      - 'bootstrap.memory_lock=true'
+      - 'cluster.name=es-mastodon'
+      - 'discovery.type=single-node'
+      - 'thread_pool.write.queue_size=1000'
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+      nofile:
+        soft: 65536
+        hard: 65536
+    restart: always
+    healthcheck:
+      test:
+        [
+          'CMD-SHELL',
+          'curl --silent --fail localhost:9200/_cluster/health || exit 1',
+        ]
+    networks:
+      - external_network
+      - internal_network
+    ports:
+      - '127.0.0.1:9200:9200'
+    volumes:
+      - ./elasticsearch:/usr/share/elasticsearch/data
 
-  ## Uncomment to enable federation with tor instances along with adding the following ENV variables
-  ## http_hidden_proxy=http://privoxy:8118
-  ## ALLOW_ACCESS_TO_HIDDEN_SERVICE=true
```

Comment:
You dropped those, they probably still need to be there. And maybe profiles could configure this automatically?